src/EPR.Calculator.API/Controllers/CalculatorInternalController.cs

```diff
@@ -153,10 +153,33 @@ public IActionResult UpdateRpdStatus([FromBody] UpdateRpdStatus request)
         [Route("prepareCalcResults")]
         public IActionResult PrepareCalcResults([FromBody] CalcResultsRequestDto resultsRequestDto)
         {
-            this.transposePomAndOrgDataService.Transpose(resultsRequestDto);
-            var results = this.builder.Build(resultsRequestDto);
-            this.exporter.Export(results);
-            return new ObjectResult(null) { StatusCode = StatusCodes.Status201Created };
+            var calculatorRun = this.context.CalculatorRuns.SingleOrDefault(run => run.Id == resultsRequestDto.RunId);
+            if (calculatorRun == null)
+            {
+                return new ObjectResult($"Unable to find Run Id {resultsRequestDto.RunId}")
+                    { StatusCode = StatusCodes.Status422UnprocessableEntity };
```

Comment:
Is 422 correct status here? I think its 404

Reply:
@Mazar-Shaik - I have fixed the code now.